src/main/java/com/powsybl/openloadflow/network/LfZeroImpedanceNetwork.java

```diff
@@ -0,0 +1,217 @@
+                    linkVoltageControls(mainVc, vc);
+                }
+            }
+        }
+    }
+
+    public void removeBranchAndTryToSplit(LfBranch disabledBranch) {
+        graph.removeEdge(disabledBranch);
+
+        List<Set<LfBus>> connectedSets = new ConnectivityInspector<>(graph).connectedSets();
+        if (connectedSets.size() > 1) { // real split
+            disabledBranch.setSpanningTreeEdge(dc, false);
+
+            Set<LfZeroImpedanceNetwork> zeroImpedanceNetworks = network.getZeroImpedanceNetworks(dc);
+            zeroImpedanceNetworks.remove(this);
+            List<LfZeroImpedanceNetwork> splitZns = new ArrayList<>(2);
+            for (Set<LfBus> connectedSet : connectedSets) {
+                var subGraph = createSubgraph(graph, connectedSet);
+                splitZns.add(new LfZeroImpedanceNetwork(network, dc, subGraph));
+            }
+            zeroImpedanceNetworks.addAll(splitZns);
+
+            // update voltage control merge status
+            if (!dc) {
+                for (LfZeroImpedanceNetwork splitZn : splitZns) {
+                    splitZn.updateVoltageControlMergeStatus();
+                }
+            }
+
+            for (LfNetworkListener listener : network.getListeners()) {
+                listener.onZeroImpedanceNetworkSplit(this, splitZns, dc);
+            }
+        } else {
+            if (disabledBranch.isSpanningTreeEdge(dc)) {
+                disabledBranch.setSpanningTreeEdge(dc, false);
+
+                // just update the spanning
+                updateSpanningTree();
+            }
+        }
+    }
+
+    public static void addBranchAndMerge(LfZeroImpedanceNetwork zn1, LfZeroImpedanceNetwork zn2, LfBranch enabledBranch) {
+        Objects.requireNonNull(zn1);
+        Objects.requireNonNull(zn2);
+        Objects.requireNonNull(enabledBranch);
+        LfNetwork network = zn1.getNetwork();
+        boolean dc = zn1.isDc();
+        Set<LfZeroImpedanceNetwork> zeroImpedanceNetworks = network.getZeroImpedanceNetworks(dc);
+        Graph<LfBus, LfBranch> mergedGraph = new Pseudograph<>(LfBranch.class);
+        Graphs.addGraph(mergedGraph, zn1.getGraph());
+        Graphs.addGraph(mergedGraph, zn2.getGraph());
+        mergedGraph.addEdge(enabledBranch.getBus1(), enabledBranch.getBus2(), enabledBranch);
+        zeroImpedanceNetworks.remove(zn1);
+        zeroImpedanceNetworks.remove(zn2);
+        LfZeroImpedanceNetwork mergedZn = new LfZeroImpedanceNetwork(network, dc, mergedGraph);
+        zeroImpedanceNetworks.add(mergedZn);
+
+        // update voltage control merge status
+        if (!dc) {
+            mergedZn.updateVoltageControlMergeStatus();
+        }
+
+        for (LfNetworkListener listener : network.getListeners()) {
+            listener.onZeroImpedanceNetworkMerge(zn1, zn2, mergedZn, dc);
+        }
+    }
+
+    public void addBranch(LfBranch branch) {
+        graph.addEdge(branch.getBus1(), branch.getBus2(), branch);
```

Comment:
@flo-dup so here we don't need to update the spanning tree anymore. Are we agree?